[System.Private.CoreLib] Compilation error: Comparing a span to 'null' might be redundant

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=589862
Build error leg or test failing: Build / browser-wasm linux Release Mono_DebuggerTests_chrome / Build product
Pull request: https://github.com/dotnet/runtime/pull/99254
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "error CA2265: Comparing a span to 'null' might be redundant, the 'null' literal will be implicitly converted to a 'Span<T>.Empty'",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```


<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=589862
**Error message validated:** `[error CA2265: Comparing a span to 'null' might be redundant, the 'null' literal will be implicitly converted to a 'Span<T>.Empty'`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 3/5/2024 6:32:50 PM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[589901](https://dev.azure.com/dnceng-public/public/_build/results?buildId=589901)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/589901/logs/55)|dotnet/runtime#99175|
|[589880](https://dev.azure.com/dnceng-public/public/_build/results?buildId=589880)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/589880/logs/63)|dotnet/runtime#99258|
|[589868](https://dev.azure.com/dnceng-public/public/_build/results?buildId=589868)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/589868/logs/101)|dotnet/runtime#99303|
|[589862](https://dev.azure.com/dnceng-public/public/_build/results?buildId=589862)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/589862/logs/56)|dotnet/runtime#99254|
|[589866](https://dev.azure.com/dnceng-public/public/_build/results?buildId=589866)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/589866/logs/71)|dotnet/runtime#99296|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|5|5|5|
<!--Known issue error report end -->

--- a/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs
@@ -344,7 +344,6 @@ namespace System.Threading
             bool waitForAll,
             int timeoutMilliseconds)
         {
-            Debug.Assert(waitHandles != null);
             Debug.Assert(waitHandles.Length > 0);
             Debug.Assert(waitHandles.Length <= WaitHandle.MaxWaitHandles);
             Debug.Assert(timeoutMilliseconds >= -1);

